Mathlib/FieldTheory/Minpoly/IsConjRoot.lean

```diff
@@ -0,0 +1,310 @@
+    exists_algHom_of_splits_of_aeval (normal_iff.mp inferInstance) (h ▸ minpoly.aeval K y)
+  exact ⟨AlgEquiv.ofBijective σ (σ.normal_bijective _ _ _), hσ⟩
+
+/--
+Let `L / K` be a normal field extension. For any two elements `x` and `y` in `L`, `y` is a
+conjugate root of `x` if and only if there exists an `K`-automorphism `σ : L ≃ₐ[K] L` such
+that `y = σ x`.
+-/
+theorem isConjRoot_iff_exists_algEquiv [Normal K L] {x y : L} :
+    IsConjRoot K x y ↔ ∃ σ : L ≃ₐ[K] L, σ x = y :=
+  ⟨exists_algEquiv, fun ⟨_, h⟩ => h ▸ isConjRoot_of_algEquiv _ _⟩
+
+/--
+Let `S / L / K` be a tower of extensions. For any two elements `y` and `x` in `S`, if `y` is a
+conjugate root of `x` over `L`, then `y` is also a conjugate root of `x` over
+`K`.
+-/
+theorem IsConjRoot.of_isScalarTower [IsScalarTower K L S] {x y : S} (hx : IsIntegral K x)
+    (h : IsConjRoot L x y) : IsConjRoot K x y :=
+  isConjRoot_of_aeval_eq_zero hx <| minpoly.aeval_of_isScalarTower K x y (aeval_eq_zero h)
+
+/--
+`y` is a conjugate root of `x` over `K` if and only if `y` is a root of the minimal polynomial of
+`x`. This is variant of `isConjRoot_iff_aeval_eq_zero`.
+-/
+theorem isConjRoot_iff_mem_aroots {x y : S} (h : IsIntegral K x) :
+    IsConjRoot K x y ↔ y ∈ (minpoly K x).aroots S := by
+  rw [Polynomial.mem_aroots, isConjRoot_iff_aeval_eq_zero h]
+  simp only [iff_and_self]
+  exact fun _ => minpoly.ne_zero h
+
+/--
+`y` is a conjugate root of `x` over `K` if and only if `y` is a root of the minimal polynomial of
+`x`. This is variant of `isConjRoot_iff_aeval_eq_zero`.
+-/
+theorem isConjRoot_iff_mem_minpoly_rootSet {x y : S}
+    (h : IsIntegral K x) : IsConjRoot K x y ↔ y ∈ (minpoly K x).rootSet S :=
+  (isConjRoot_iff_mem_aroots h).trans (by simp [rootSet])
+
+namespace IsConjRoot
+
+/--
+If `y` is a conjugate root of an integral element `x` over `R`, then `y` is also integral
+over `R`.
+-/
+theorem isIntegral {x y : A} (hx : IsIntegral R x) (h : IsConjRoot R x y) :
+    IsIntegral R y :=
+  ⟨minpoly R x, minpoly.monic hx, h ▸ minpoly.aeval R y⟩
+
+/--
+A variant of `IsConjRoot.eq_of_isConjRoot_algebraMap`, only assuming `Nontrivial R`,
+`NoZeroSMulDivisors R A` and `Function.Injective (algebraMap R A)` instead of `Field R`. If `x` is a
+conjugate root of some element `algebraMap R S r` in the image of the base ring, then
+`x = algebraMap R S r`.
+-/
+theorem eq_algebraMap' [Nontrivial R] [NoZeroSMulDivisors R S] {r : R} {x : S}
+    (h : IsConjRoot R x (algebraMap R S r)) (hf : Function.Injective (algebraMap R S)) :
+    x = algebraMap R S r := by
+  rw [IsConjRoot, minpoly.eq_X_sub_C_of_algebraMap_inj _ hf] at h
+  have : x ∈ (X - C r).aroots S := by
+    rw [mem_aroots]
+    simp [X_sub_C_ne_zero, h ▸ minpoly.aeval R x]
+  simpa [aroots_X_sub_C] using this
+
+/--
+If `x` is a conjugate root of some element `algebraMap R S r` in the image of the base ring, then
+`x = algebraMap R S r`.
+-/
+theorem eq_algebraMap {r : K} {x : S} (h : IsConjRoot K x (algebraMap K S r)) :
```

Comment:
Maybe add a `@[simp]`-tagged `iff` version for this, like you did for `isConjRoot_zero_iff_eq_zero`?

Reply:
Thank you! Added.